### ISSUE TITLE
[experiment] Service-based self-polling watchbot with containers that exit after every job

### ISSUE THREAD

**Issue by jakepruitt** [mapbox/ecs-watchbot#190]
## The background

Helpful reading:

- ♻️ that container pull request: https://github.com/mapbox/ecs-watchbot/pull/184
- Disk monitoring and quotas: https://github.com/mapbox/ecs-watchbot/pull/188
  - "With the potential risk of one watchbot stack taking down an entire cluster of EC2 disk space, we can't merge #184 without first having some sort of safeguard in place."

The current implementation of #184 "♻️ that container" runs the risk of filling disk space on ECS hosts as successive watchbot jobs are run inside the same container. In order to prevent the hosts from filling up with the leftover files of every job, we could potentially try a few different options:

- Stop containers after every job, thereby creating a fresh new environment for every job to run in. This defeats some of the goals of the "♻️ that container" pull request, but it still leaves us in a place that doesn't rely on the RunTask API. It will instead rely on the service scheduler, which may or may not have the same underlying throughput problems as the RunTask API. We cannot know the behavior until we test it out.
- Build code into the watcher that restricts the write access of the worker subprocess to just the docker volumes, then wipe the docker volumes clean after every job.

This ticket outlines the plan for the first branch of the experiment: making every container exit after every job. 

## The plan

We should start with the #184 "♻️ this container" branch's code. @rclark has already done the legwork of writing code for running watchbot as an ECS service, polling the SQS queue from within the container, and running the main script as a subprocess.

The code we'll want to change is specifically around the way the watcher handles the worker's success - we'll want to either `process.exit()` the main loop after the worker completes, or do something that's easier to mock/test. I think doing `process.exit()` is probably fine for now, and if we decide this is the direction we want to pursue, we can refine the code a bit more.

## The next actions

Basically, our next actions are to implement the plan. I think this can be done with some specific code changes:

- [ ] Make a branch from the #184 "♻️ this container" branch.
- [ ] Modify the `watcher.js` code to exit the main node process after workers complete.
- [ ] Implement tests that the process exits: there may be some tough mocking to make this easily testable.
- [ ] Confirm the new watchbot version works by testing out with https://github.com/mapbox/ecs-telephone.
- [ ] Load test the new watchbot version with a really high concurrency load of very fast tasks.

cc/ @mapbox/platform-engine-room

**Comment by arunasank** [comment]
Closed via https://github.com/mapbox/ecs-watchbot/pull/196
